boards: share the leftover mcuboot scratch flash between image 0 and image 1 for nRF devices

nvlsianpu opened the pull request:
Newest mcuboot uses move swap instead of scratch
for nRF devices

Share the leftover scratch flash between image 0 and image 1.
For trustzone enabled devices, give the non secure
partitions the extra space.

Adjusted test overlays to above changes.

[EDIT]
removed manifest upgrade from the patch content
manifest upgrade is here #22369 
merge after that PR.

[EDIT]
For nrf5340pdk_nrf5340_cpuapp and nrf9160dk_nrf9160 targets
additionally cortected partitions bounduaries as after
the boot partition was resize
#25874
a few flash sectors were left unused.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>
Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

hakonfam commented:
How should this be reflected in the dts files which currently set aside a rather large amount of space for the scratch area?

hakonfam reviewed (changes requested):
Need clarification on how to remove scratch partition from DTS.

nvlsianpu commented:
I believe we can remove scratch partition from Nordic boards DTS even within this patch.

hakonfam commented:
Please cherry-pick the changes here https://github.com/hakonfam/zephyr/tree/remove-scratch-nrf 

It removed the scratch partition for all nrf boards

nvlsianpu commented:
^^ @hakonfam  done.

hakonfam committed:
dts: remove scratch area from nrf boards

The default configuration for mcuboot with nrf is to use
move based swap instead of scratch based.

Share the leftover flash between image 0 and image 1.
For trustzone enabled devices, give the non secure
partitions the extra space.

For nrf5340pdk_nrf5340_cpuapp and nrf9160dk_nrf9160 targets
additionally cortected partitions bounduaries as after
the boot partition was resize
https://github.com/zephyrproject-rtos/zephyr/pull/25874
a few flash sectors were left unused.

Adjusted test overlays to above changes.

Signed-off-by: Håkon Øye Amundsen <haakon.amundsen@nordicsemi.no>

mbolivar commented:
Can't wait for the support requests, should be interesting ;)

hakonfam commented:
@nvlsianpu  I updated the branch with resizing the other partitions to re-use the now freed flash.

nvlsianpu commented:
@hakonfam Updated, just kept comments on storage partition purpose.

nvlsianpu committed:
tests/subsys/dfu/mcuboot: fix DT overlay

Overlay for nrf52840_pca10056 board was going to overlay not
deleted previously nodes which cause DTC error in case original
nodes were different.

Signed-off-by: Andrzej Puzdrowski <andrzej.puzdrowski@nordicsemi.no>

carlescufi commented on west.yml:
```diff
@@ -77,7 +77,7 @@ manifest:
       path: modules/crypto/mbedtls
     - name: mcuboot
-      revision: 54c1e3fb6ba4dcf7b315943bcfcda1e10a4042f5
+      revision: 09be78364430f48287fad54ff66c16485c93d372
```
Another PR was merged, this needs updating

pabigot replied:
Agreed; this is currently two behind master.

pabigot commented:
Actually this should be closed as superseded by #22369 as far as I can tell.

nvlsianpu commented:
@carlescufi @pabigot Removed the manifest update commit. Redacted the PR title to more relevant.

ioannisg reviewed (commented):
needs rebase now @nvlsianpu

ioannisg commented:
Wait until all nRF boards are renamed

nvlsianpu commented:
> Wait until all nRF boards are renamed
Will wait with upgrade this patch until reaming will be done.

de-nordic reviewed (approved):
Looks fine.

ioannisg reviewed (changes requested):
Before merging this I 'd like to resolve how to align with the SPU granularity, at least for SoCs that have the SPU.

carlescufi commented:
@nvlsianpu can you please rebase?
